sklearnex/neighbors/knn_regression.py

```diff
@@ -1,18 +1,393 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+try:
+    from packaging.version import Version
+except ImportError:
+    from distutils.version import LooseVersion as Version
+from sklearn import __version__ as sklearn_version
+import warnings
+
+from sklearn.neighbors._base import NeighborsBase as sklearn_NeighborsBase
+from sklearn.neighbors._ball_tree import BallTree
+from sklearn.neighbors._kd_tree import KDTree
+from sklearn.neighbors._base import _check_weights
+from sklearn.neighbors._base import VALID_METRICS
+from sklearn.neighbors._regression import KNeighborsRegressor as \
+    sklearn_KNeighborsRegressor
+from sklearn.neighbors._unsupervised import NearestNeighbors as \
+    sklearn_NearestNeighbors
+from sklearn.utils.validation import _deprecate_positional_args, check_is_fitted
+
+from onedal.datatypes import _check_array, _num_features, _num_samples
+from onedal.neighbors import KNeighborsRegressor as onedal_KNeighborsRegressor
+
+from .._device_offload import dispatch, wrap_output_data
+import numpy as np
+from scipy import sparse as sp
+
+
+if Version(sklearn_version) >= Version("0.24"):
+    class KNeighborsRegressor_(sklearn_KNeighborsRegressor):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = \
+                weights if Version(sklearn_version) >= Version("1.0") \
+                else _check_weights(weights)
+elif Version(sklearn_version) >= Version("0.22"):
+    from sklearn.neighbors._base import SupervisedFloatMixin as \
+        BaseSupervisedFloatMixin
+
+    class KNeighborsRegressor_(sklearn_KNeighborsRegressor,
+                               BaseSupervisedFloatMixin):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+else:
```

Comment:
Do I understand correctly that this code branch is for version <= 0.21?
We support 4 latest versions of sklearn. At the moment it is 1.0, 0.24, 0.23, 0.22